actions/utils.go

```diff
@@ -46,3 +59,21 @@ func join(A []string, delim string) string {
 
 	return buffer.String()
 }
+
+// Randomly generate a set of indexes in each sector
+func generateInsertedIndexesForPearl(fileSizeInByte int) []int {
+	if fileSizeInByte <= 0 {
+		return nil
+	}
+
+	numOfSectors := int(math.Ceil(float64(fileSizeInByte) / fileSectorInByte))
+	remainderOfChunks := math.Ceil(float64(fileSizeInByte) / fileChunkSizeInByte)
```

Comment:
I think we should return an array of ints `i` where `0 <= i < sectorChunksCount`, each sector would be able to offset it knowing which sector it's in. So we don't need to know the remainder.

For example, and array `[5, 1]` and sector size of `5` would mean the first treasure is inserted at `3` and second inserted at `11`

We just need to be careful about the last sector, which might not be complete.

Reply:
The problem is that when we insert the pearl, it would push one file chunk to the next sector, thus, all the remaining file chunk will need to shift down.

Reply:
If you have 102 chunks, and 10 chunks per sector, you will know that you will have 11 treasure chunks 
```ceil( 102 / 10 ) = 11```

From this you know there will be 113 total chunks
```102+11```

So your last treasure index will be i where `0 <= i < 3`

So a sample return value is `[ 6, 2, 7, ..., 2 ]`

Reply:
Ok, I will remove the offset then